src/take.test.ts

```diff
@@ -16,3 +24,139 @@ describe("data_last", () => {
+    expectTypeOf(take(restTail, 2)).toEqualTypeOf<[string, ...Array<number>]>();
+
+    const restTailReadonly = ["1", 2, 3] as readonly [string, ...Array<number>];
+    expectTypeOf(take(restTailReadonly, 1)).toEqualTypeOf<[string]>();
+    expectTypeOf(take(restTailReadonly, 2)).toEqualTypeOf<
+      [string, ...Array<number>]
+    >();
+  });
+
+  it("handles rest heads", () => {
+    const restHead = [1, 2, 3, "foo"] as [...Array<number>, string];
+    expectTypeOf(take(restHead, 3)).toEqualTypeOf<[...Array<number>, string]>();
+
+    const restHeadReadonly = [1, 2, 3, "foo"] as readonly [
+      ...Array<number>,
+      string,
+    ];
+    expectTypeOf(take(restHeadReadonly, 3)).toEqualTypeOf<
+      [...Array<number>, string]
+    >();
+  });
+
+  it("handles a mix of unions and rest params", () => {
+    const mix = ["cat", 123, 123, 456] as ["cat" | "dog", ...Array<123 | 456>];
+    expectTypeOf(take(mix, 1)).toEqualTypeOf<["cat" | "dog"]>();
+    expectTypeOf(take(mix, 3)).toEqualTypeOf<
+      ["cat" | "dog", ...Array<123 | 456>]
+    >();
+
+    const mixReadonly = ["cat", 123, 123, 456] as readonly [
+      "cat" | "dog",
+      ...Array<123 | 456>,
+    ];
+    expectTypeOf(take(mixReadonly, 1)).toEqualTypeOf<["cat" | "dog"]>();
+    expectTypeOf(take(mixReadonly, 3)).toEqualTypeOf<
+      ["cat" | "dog", ...Array<123 | 456>]
+    >();
+  });
+
+  it("handles N greater than tuple length", () => {
+    const short = ["foo"] as [string];
+    expectTypeOf(take(short, 10)).toEqualTypeOf<[string]>();
+
+    const shortReadonly = ["foo"] as readonly [string];
+    expectTypeOf(take(shortReadonly, 10)).toEqualTypeOf<[string]>();
+  });
+
+  it("handles shorter heads", () => {
+    const shortHead = [1, "foo", true, false] as [
+      number,
+      string,
+      ...Array<boolean>,
+    ];
+    expectTypeOf(take(shortHead, 3)).toEqualTypeOf<
+      [number, string, ...Array<boolean>]
+    >();
+
+    const shortHeadReadonly = [1, "foo", true, false] as readonly [
+      number,
+      string,
+      ...Array<boolean>,
+    ];
+    expectTypeOf(take(shortHeadReadonly, 3)).toEqualTypeOf<
+      [number, string, ...Array<boolean>]
+    >();
+  });
+
+  it("handles longer heads", () => {
+    const longHead = [1, "foo", true, "foo", "bar"] as [
+      number,
+      string,
+      boolean,
+      ...Array<string>,
+    ];
+
+    expectTypeOf(take(longHead, 2)).toEqualTypeOf<[number, string]>();
+    expectTypeOf(take(longHead, 6)).toEqualTypeOf<
+      [number, string, boolean, ...Array<string>]
+    >();
+
+    const longHeadReadonly = [1, "foo", true, "foo", "bar"] as readonly [
+      number,
+      string,
+      boolean,
+      ...Array<string>,
+    ];
+
+    expectTypeOf(take(longHeadReadonly, 2)).toEqualTypeOf<[number, string]>();
+    expectTypeOf(take(longHeadReadonly, 6)).toEqualTypeOf<
+      [number, string, boolean, ...Array<string>]
+    >();
+  });
+
+  it("handles regular arrays", () => {
+    const input = [1, 2, 3];
```

Comment:
1. cast this to an array, i'm not sure the type isn't a tuple here...
2. you don't need values for type tests, only types...
```suggestion
    const input = [] as Array<number>;
```